src/audio/dai.c

```diff
@@ -235,6 +235,26 @@ static int dai_comp_get_hw_params(struct comp_dev *dev,
 	return 0;
 }
 
+static int dai_verify_params(struct comp_dev *dev,
+			     struct sof_ipc_stream_params *params)
+{
+	struct sof_ipc_stream_params hw_params;
+
+	dai_comp_get_hw_params(dev, &hw_params);
+	if ((hw_params.rate != params->rate && hw_params.rate) ||
+	    (hw_params.channels != params->channels && hw_params.channels)) {
```

Comment:
Just need to split this logic for all conditional config combinations.
```
if (rate != hw->rate)
    trace_err("error: rate .....);
if (format != hw->format)
     trace_err("error: format .....);
```
Please do this for all combined configuration logic, making each check individual means it's easier to debug and integrate with the different traces and the code is also easier to read.

Reply:
@lgirdwood 
I've added error returning in ```comp_dai_get_hw_params()```.
and made individual (for each parameter) checks in ```dai_verify_params()```.